priority_directives_map.go

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package crossplane
+
+// A map used to overwrite generated directive definitions.
+// When we have a human-defined definition for a directive, which is different
+// from the definition in source code, put it here.
+//
+//nolint:gochecknoglobals
+var priorityMap = map[string][]uint{
+	"if": {
+		ngxHTTPSrvConf | ngxHTTPLocConf | ngxConfBlock | ngxConfExpr | ngxConf1More,
+	},
+}
```

Comment:
How about something like this. I've renamed the variable to match it's intended use and included detail from your [comment](https://github.com/nginxinc/nginx-go-crossplane/pull/100#discussion_r1655444060) about the `if` directive.

```go
// directiveOverrides is a map of directive names to masks used to override any
// masks that are provided to tell the parser how to validate directives and their
// arguments.
//nolint:gochecknoglobals
var directiveOverrides = map[string][]uint{
	// if contains a bitmask (ngxConfExpr) that does not exist in NGINX that indicates
	// the directive name is followed by an expression in parantheses.
	"if": {
		ngxHTTPSrvConf | ngxHTTPLocConf | ngxConfBlock | ngxConfExpr | ngxConf1More,
	},
}
```

Reply:
I'm wondering if this has to obliterate any masks passed by the caller, or if this should instead be appended to the list of masks? It may be unlikely, but what if another module provides an `if` directive to be used in other contexts?

Reply:
> How about something like this. I've renamed the variable to match it's intended use and included detail from your https://github.com/nginxinc/nginx-go-crossplane/pull/100#discussion_r1655444060 about the if directive.

Sure! It seems more clear. And the comments for the reason we have `if` here makes it much better.

> I'm wondering if this has to obliterate any masks passed by the caller, or if this should instead be appended to the list of masks?

If we apply "append to DirectiveSources", the `if` statement can pass the validation even if no `()` after it. The reason is current validation logic for bitmasks is `or`. In "append to DirectiveSources", `if` can pass validation through the bitmask from default map(or `DirectiveSources`). By overriding user has high flexibility, if in the future a new bitmask of `if` appears in OSS, user can add the new bitmask to `directiveOverrides`.

I've thought of providing a "overrideMap" and a "appendMap" before, but it seems the "appendMap" is unnecessary since user can put all bitmasks in overrideMap to achieve its function.

> It may be unlikely, but what if another module provides an if directive to be used in other contexts?

I think this situation it's kind of out of current scope? To address this we need to define a "override map" for every module, when the module is included, the corresponding "override map"  should be included together. But it seems very unlikely to happen since we only have a case of `if` till now.

Reply:
Say I implement my own `ngx_foo.so` module that adds a new context and with a new directive called `if` that implements a different syntax.

```nginx
foo {
    if $arg_foo == "";
}
```

How can I as a module author provide a `MatchFunc` to handle my new directive that occurs in the `foo` context? I am less concerned about `if` specifically and more concerned about future expansion.